src/main/java/entities/Post.java

```diff
@@ -43,6 +43,13 @@ public Post(User poster, String title, String mainDesc, List<String> tags, Strin
         this.favouritedUsers = new ArrayList<>();
     }
 
```

Comment:
It looks like your entities are not up to date with the main branch so this code won't work.